src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,29 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() ->
+                cache.computeIfAbsent(userId, id -> {
+                    try {
+                        // Simulate a delay for fetching user data
+                        TimeUnit.SECONDS.sleep(1);
```

Comment:
Sleeping in the computeIfAbsent method will block the cache for other threads accessing this userId. This can lead to performance issues if multiple threads are trying to access the cache for different user IDs at the same time. Consider fetching the data in a way that doesn't block the cache, such as using a separate service or method that handles data retrieval without sleep.